Dockerfile-windows.template

```diff
@@ -0,0 +1,57 @@
+FROM mcr.microsoft.com/windows/servercore:version as installer
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+# PATH isn't actually set in the Docker image, so we have to set it from within the container
+RUN $newPath = ('C:\Program Files (x86)\GnuPG\bin;{0}' -f $env:PATH); \
+    Write-Host ('Updating PATH: {0}' -f $newPath); \
+    [Environment]::SetEnvironmentVariable('PATH', $newPath, [EnvironmentVariableTarget]::Machine)
+# doing this first to share cache across versions more aggressively
+
+ENV NODE_VERSION 0.0.0
+ENV NODE_CHECKSUM CHECKSUM_x64
+
+# Version and checksum of the GPG installer (Source: https://www.gnupg.org/download/integrity_check.html)
+ENV GPG_VERSION 2.4.5_20240307
+ENV GPG_CHECKSUM d2ac821ceacf9409ebcdb42ae330087ada30c732981f00b356f9c2f08fac4dc1
+
+RUN Invoke-WebRequest $('https://www.gnupg.org/ftp/gcrypt/binary/gnupg-w32-{0}.exe' -f $env:GPG_VERSION) -OutFile 'gpg-installer.exe'; \
+    if ((Get-FileHash gpg-installer.exe -Algorithm sha256).Hash -ne $env:GPG_CHECKSUM) { Write-Error 'GPG checksum mismatch' }; \
+    Start-Process -FilePath 'gpg-installer.exe' -ArgumentList '/S' -Wait; \
+    gpg --version;
```

Comment:
I'm not 100% sold that the benefit of verifying this signature during build (especially if we verify the `NODE_CHECKSUM` as I've noted above) warrants the added downsides of a multi-stage build. :thinking:

(https://github.com/docker-library/faq#multi-stage-builds)

Reply:
The multi-stage build isn't here just because of gpg, it's also because nano doesn't support PowerShell, based on what I've read and tested. Essentially, it's much easier to download and verify files within a Server Core server than in nano. We could potentially copy the files from one stage to another within nano if we wanted to add a nano server variant. However, we couldn't use PowerShell in that variant, so we'd need to find an alternative way to update the environment variable for example.